specification/dsdl/serialization.tex

```diff
@@ -77,12 +82,81 @@ \subsubsection{Implicit padding}
 handed over for transmission over the UAVCAN network
 (or any other destination which does not support bit-level data granularity).
 
-Additionally, the transport layer may introduce extra padding at the end of the serialized representation,
-as described in relevant sections of chapter \ref{sec:transport_layer};
-however, such padding behavior falls outside of the scope of the DSDL specification
-as it belongs to the domain of network transports rather than data presentation.
+\subsubsection{Implicit truncation of excessive data}
+
+When a serialized representation is deserialized, implementations shall silently\footnote{%
+    I.e., without registering an error of any kind.
+} ignore any excessive (unused) data remaining upon deserialization, irrespective of its amount.
+
+One of the consequences of the above requirement is that the transport layer will be able to introduce
+additional padding at the end of the serialized representation to satisfy its data size granularity constraints.
+The specifics of such padding behavior fall outside of the scope of the DSDL specification
+as they belong to the domain of network transports rather than data presentation.
+
+Another consequence is that a serialized representation constructed from an instance of type $B$ can be
+deserialized into an instance of type $A$ as long as $B$ is a structural subtype of $A$.
+The topic of data type compatibility is explored in section \ref{sec:dsdl_versioning}.
+
+\begin{remark}
+    Let $X$ be an instance of data type $B$, which is defined as follows:
+
+    \begin{minted}{python}
+        float32 parameter
+        float32 variance
+    \end{minted}
+
+    Let $A$ be a structural supertype of $B$, being defined as follows:
+
+    \begin{minted}{python}
+        float32 parameter
+    \end{minted}
+
+    Then the serialized representation of $X$ can be deserialized into an instance of $A$.
+
+    The implicit truncation rule is introduced to permit extension of data types by introducing additional fields
+    without breaking backward compatibility with existing deployments,
+    and to permit introduction of more specific derived types while retaining compatibility with their base types.
+    The topic of data type compatibility is explored in detail in section \ref{sec:dsdl_versioning}.
+\end{remark}
 
-When a serialized representation is deconstructed, the value of the trailing padding bits must be ignored.
+\subsubsection{Implicit zero extension of missing data}
+
+For the purposes of deserialization routines,
+the serialized representation of any instance of any data type shall be considered to contain an implicit
+sequence of trailing zero (0) bits of infinite length.
+
+Despite this rule, implementations are not allowed to intentionally truncate trailing zeros
+upon construction of a serialized representation of an object\footnote{%
+    Intentional truncation is prohibited because in a future revision of the specification the implicit zero
```

Comment:
```suggestion
    Intentional truncation is prohibited because in a future, major revision of the specification the implicit zero
```

Reply:
It doesn't have to be major because this change would not violate backward compatibility with existing systems.

Reply:
Okay.